native/service/cross_chain_manager/cosmos/cosmos_handler.go

```diff
@@ -73,9 +56,8 @@ func (this *CosmosHandler) MakeDepositProposal(service *native.NativeService) (*
 	if len(params.HeaderOrCrossChainMsg) == 0 {
 		return nil, fmt.Errorf("you must commit the header used to verify transaction's proof and get none")
 	}
-	cdc := newCDC()
 	var myHeader cosmos.CosmosHeader
-	if err := cdc.UnmarshalBinaryBare(params.HeaderOrCrossChainMsg, &myHeader); err != nil {
+	if err := cosmos.Cdc.UnmarshalBinaryBare(params.HeaderOrCrossChainMsg, &myHeader); err != nil {
```

Comment:
it's safer to only change cdc for new version, but keep it the same for old version

Reply:
I have changed the codec to use v33, and only use v34 encoding when block version is detected to be new.